geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -1099,6 +1099,30 @@ public void lockDiskStore(String diskStoreName) {
     doLockDiskStore(diskStoreName);
   }
 
+  /**
+   * Having this method to return a SecurityServiceFactory allows users to replace the factory
+   * for testing purposes.
+   *
+   * In a test set the system property "org.apache.geode.internal.security.SecurityServiceFactory"
+   * to the class of choice ie. [Classname].class.getName() and in this class either implement
+   * SecurityServiceFactory or override the default implementation to create the desired behavior.
```

Comment:
I'm really not a fan of the idea of introducing a system property solely to support testing. Is there any other approach that could be used here?

Reply:
@DonalEvans I agree with that but unfortunately the code is such that I have not been able to come up with an alternative approach.